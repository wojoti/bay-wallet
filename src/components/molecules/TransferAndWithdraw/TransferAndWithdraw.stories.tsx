import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {TransferAndWithdraw} from 'components/molecules';

export default {
  title: 'Molecules/TransferAndWithdraw',
  component: TransferAndWithdraw,
  argTypes: {
    onTransferPress: {action: 'onTransferPress'},
    onWithdrawPress: {action: 'onWithdrawPress'},
  },
} as ComponentMeta<typeof TransferAndWithdraw>;
const Template: ComponentStory<typeof TransferAndWithdraw> = args => (
  <TransferAndWithdraw {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
};
