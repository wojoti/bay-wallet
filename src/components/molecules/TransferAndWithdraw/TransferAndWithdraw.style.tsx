import {StyleSheet} from 'react-native';
import {lightPrimary} from 'theme/colors';

const style = StyleSheet.create({
  btnStyle: {
    height: 60,
    backgroundColor: lightPrimary,
    borderRadius: 14,
  },
  sizer: {
    width: 155,
  },
});
export default style;
