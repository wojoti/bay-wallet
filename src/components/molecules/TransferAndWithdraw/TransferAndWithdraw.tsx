import {
  ComposableButton,
  Container,
  Img,
  ImgType,
  Label,
} from 'components/atoms';
import {PropsWithChildren} from 'react';
import {secondary} from 'theme/colors';
import style from './TransferAndWithdraw.style';

export type TransferAndWithdrawProps = PropsWithChildren<{
  onTransferPress: () => void;
  onWithdrawPress: () => void;
  testId?: string;
}>;

const TransferAndWithdraw = ({
  onTransferPress,
  onWithdrawPress,
  testId,
}: TransferAndWithdrawProps) => {
  return (
    <Container direction="row" justify="space-between" testId={testId}>
      <ComposableButton
        customStyle={style.btnStyle}
        onPress={onTransferPress}
        testId={'transfer-button'}>
        <Container
          customStyle={style.sizer}
          direction="row"
          justify="space-evenly"
          align="flex-start">
          <Img src={ImgType.transfer} height={'24'} width={24} alt={''} />
          <Label size={15} color={secondary}>
            Transfer
          </Label>
        </Container>
      </ComposableButton>
      <ComposableButton
        customStyle={style.btnStyle}
        onPress={onWithdrawPress}
        testId={'withdraw-button'}>
        <Container
          customStyle={style.sizer}
          direction="row"
          justify="space-evenly"
          align="flex-start">
          <Img src={ImgType.withdraw} height={'24'} width={24} alt={''} />
          <Label size={15} color={secondary}>
            Withdraw
          </Label>
        </Container>
      </ComposableButton>
    </Container>
  );
};

export default TransferAndWithdraw;
