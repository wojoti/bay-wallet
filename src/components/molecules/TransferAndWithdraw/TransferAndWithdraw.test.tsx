import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import TransferAndWithdraw, {
  TransferAndWithdrawProps,
} from './TransferAndWithdraw';

const mockOnTransferPress = jest.fn();
const mockOnWithdrawPress = jest.fn();
const props: TransferAndWithdrawProps = {
  testId: 'test-transferandwithdraw-id',
  onTransferPress: mockOnTransferPress,
  onWithdrawPress: mockOnWithdrawPress,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<TransferAndWithdraw {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render TransferAndWithdraw', () => {
  render(<TransferAndWithdraw {...props} />);
  const TransferAndWithdrawElement = screen.getByTestId(
    'test-transferandwithdraw-id',
  );
  expect(TransferAndWithdrawElement).toBeOnTheScreen();
});
