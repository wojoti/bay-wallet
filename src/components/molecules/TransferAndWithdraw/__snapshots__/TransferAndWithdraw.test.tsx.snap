// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": undefined,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="test-transferandwithdraw-id"
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderStyle": "solid",
          "justifyContent": "center",
          "opacity": 1,
        },
        false,
        {
          "backgroundColor": "#4E439E",
          "borderRadius": 14,
          "height": 60,
        },
      ]
    }
    testID="transfer-button"
  >
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "width": "100%",
          },
          {
            "width": 155,
          },
        ]
      }
    >
      <SvgMock
        height="24"
        width={24}
      />
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Poppins-Light",
            "fontSize": 15,
            "fontWeight": "400",
            "textAlign": "left",
          }
        }
      >
        Transfer
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderStyle": "solid",
          "justifyContent": "center",
          "opacity": 1,
        },
        false,
        {
          "backgroundColor": "#4E439E",
          "borderRadius": 14,
          "height": 60,
        },
      ]
    }
    testID="withdraw-button"
  >
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "width": "100%",
          },
          {
            "width": 155,
          },
        ]
      }
    >
      <SvgMock
        height="24"
        width={24}
      />
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Poppins-Light",
            "fontSize": 15,
            "fontWeight": "400",
            "textAlign": "left",
          }
        }
      >
        Withdraw
      </Text>
    </View>
  </View>
</View>
`;
